tests/integrationv2/test_record_padding.py

```diff
@@ -0,0 +1,104 @@
+import copy
+import pytest
+
+from configuration import available_ports, ALL_TEST_CURVES, MINIMAL_TEST_CERTS
+from common import Ciphers, ProviderOptions, Protocols, data_bytes
+from fixtures import managed_process  # lgtm [py/unused-import]
+from providers import Provider, S2N, OpenSSL
+from utils import invalid_test_parameters, get_parameter_name, get_expected_s2n_version, to_bytes
+
+PADDING_SIZE_MIN = 0
+PADDING_SIZE_SMALL = 250
+PADDING_SIZE_MEDIUM = 1000
+PADDING_SIZE_MAX = 1 << 14
+
+PADDING_SIZES = [
+    PADDING_SIZE_MIN,
+    PADDING_SIZE_SMALL,
+    PADDING_SIZE_MEDIUM,
+    PADDING_SIZE_MAX
+]
+
+PAYLOAD_SIZE_SMALL = 10
+PAYLOAD_SIZE_MEDIUM = 100
+# openssl has a max fragment length of 4096 bytes
+PAYLOAD_SIZE_LARGE = 4096
+
+OPENSSL_PAYLOAD_SIZES = [PAYLOAD_SIZE_SMALL,
+                         PAYLOAD_SIZE_MEDIUM, PAYLOAD_SIZE_LARGE]
+
+# https://www.rfc-editor.org/rfc/rfc5116
+TLS13_CIPHERS_WITH_16_BYTE_TAGS = [
+    Ciphers.AES128_GCM_SHA256,
+    Ciphers.AES256_GCM_SHA384,
+]
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", TLS13_CIPHERS_WITH_16_BYTE_TAGS, ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
```

Comment:
Why is this test restricted to TLS13?

Reply:
The original ticket this pr is addressing specified tls13: https://github.com/aws/s2n-tls/issues/1615. I *think* the padding behaviour is similar in tls 1.2 but I didn't get a chance to go through those requirements

Reply:
Looks like record padding was added in TLS1.3 and doesn't exist in earlier versions. I would add a comment here mentioning that record padding is a TLS1.3 feature.